controllers/host/host_controller.go

```diff
@@ -1590,16 +1586,47 @@ func (r *HostReconciler) ReconcileExistingHost(client *gophercloud.ServiceClient
 		SyncIFNameByUuid(profile, current)
 	}
 
-	err = r.ReconcilePlatformNetworks(client, instance, profile, &hostInfo)
+	host_instance, system_info, err := r.CloudManager.GetActiveHostAndSystemInfo(request_namespace, client)
 	if err != nil {
-		return err
+		msg := "failed to get active host and system info"
+		logHost.Error(err, msg)
+		return common.NewUserDataError(msg), platform_network_subreconciler_errs
+	}
+
+	if instance.Name == host_instance.Name {
+		// Only reconcile platform networks in active controller reconciliation
+		// loops to prevent concurrency issues.
+		// Also within active controller prevent potential concurrent reconciliation
+		// if IsPlatformNetworkReconciling.
+		if !r.CloudManager.IsPlatformNetworkReconciling() {
+			r.CloudManager.SetPlatformNetworkReconciling(true)
+
+			// Do not allow notifying active host until ReconcilePlatformNetworks
+			// is completed, this is to prevent unnecessary status update conflicts
+			// in addrpools and platform network resources.
+			r.CloudManager.SetNotifyingActiveHost(true)
+
+			platform_network_subreconciler_errs = r.ReconcilePlatformNetworks(client, instance, profile, &hostInfo, system_info)
+
+			r.CloudManager.SetPlatformNetworkReconciling(false)
+			r.CloudManager.SetNotifyingActiveHost(false)
+
+			if len(platform_network_subreconciler_errs) != 0 {
+				return nil, platform_network_subreconciler_errs
+			}
+
+		} else {
+			err_msg := "platform network reconciliation is already in progress in active controller."
```

Comment:
As it is expected to only update in the active controller, can we change the message to "waiting for platform network reconciled"?